spring-cloud-azure-feature-management/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/FeatureManager.java

```diff
@@ -64,46 +65,48 @@ public FeatureManager(FeatureManagementConfigProperties properties) {
      * 
      * @param feature Feature being checked.
      * @return state of the feature
-     * @throws FilterNotFoundException 
+     * @throws FilterNotFoundException
      */
     public Mono<Boolean> isEnabledAsync(String feature) throws FilterNotFoundException {
         return Mono.just(checkFeatures(feature));
     }
 
     private boolean checkFeatures(String feature) throws FilterNotFoundException {
-        boolean enabled = false;
         if (featureManagement == null || onOff == null) {
             return false;
         }
 
-        Feature featureItem = featureManagement.get(feature);
         Boolean boolFeature = onOff.get(feature);
 
         if (boolFeature != null) {
             return boolFeature;
-        } else if (featureItem == null) {
+        }
+
+        Feature featureItem = featureManagement.get(feature);
+        if (featureItem == null) {
             return false;
         }
 
-        for (FeatureFilterEvaluationContext filter : featureItem.getEnabledFor().values()) {
-            if (filter != null && filter.getName() != null) {
-                try {
-                    FeatureFilter featureFilter = (FeatureFilter) context.getBean(filter.getName());
-                    enabled = Mono.just(featureFilter.evaluate(filter)).block();
-                } catch (NoSuchBeanDefinitionException e) {
-                    LOGGER.error("Was unable to find Filter " + filter.getName()
-                            + ". Does the class exist and set as an @Component?");
-                    if (properties.isFailFast()) {
-                        String message = "Fail fast is set and a Filter was unable to be found";
-                        ReflectionUtils.rethrowRuntimeException(new FilterNotFoundException(message, e, filter));
-                    }
-                }
-            }
-            if (enabled) {
-                return enabled;
+        return featureItem.getEnabledFor().values().stream().filter(Objects::nonNull)
+                .filter(featureFilter -> featureFilter.getName() != null)
+                .map(featureFilter -> isFeatureOn(featureFilter, feature)).findAny().orElse(false);
+    }
+
+    private boolean isFeatureOn(FeatureFilterEvaluationContext filter, String feature) {
+        try {
+            FeatureFilter featureFilter = (FeatureFilter) context.getBean(filter.getName());
+            filter.setFeatureName(feature);
+
+            return Mono.just(featureFilter.evaluate(filter)).block();
+        } catch (NoSuchBeanDefinitionException e) {
+            LOGGER.error("Was unable to find Filter " + filter.getName()
+                    + ". Does the class exist and set as an @Component?");
```

Comment:
It's recommended for this string.format to use the logger feature:
```suggestion
LOGGER.error("Was unable to find Filter {}. Does the class exist and set as an @Component?", filter.getName());
```